src/main/java/com/sps/Servlets/NewRecipeServlet.java

```diff
@@ -91,29 +93,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     recipe.setProperty("ingredients", ingredients);
     recipe.setProperty("steps", steps);
     recipe.setProperty("search-strings", new ArrayList<String>(searchStrings));
+    recipe.setProperty("timestamp", timestamp);
     datastore.put(recipe);
 
     response.sendRedirect("/edit-recipe.html");
   }
 
   /**
    * Gets the parameters for fields that have different numbers of parameters from recipe to recipe.
-   * For example, one recipe may have 2 ingredients, while another may have 20.
    * This method ensures that all tags, ingredients, and steps are recorded, no matter how many of each a recipe has.
+   * @return A Collection of EmbeddedEntities, where each EmbeddedEntity holds one parameter.
    */
-  private Collection<EmbeddedEntity> getParameters(HttpServletRequest request, String type, Collection<String> searchStrings) {
+  private Collection<EmbeddedEntity> getParameters(HttpServletRequest request, String field, Collection<String> searchStrings) {
```

Comment:
Couldn't, and shouldn't, you use request.getParameterMap() instead of this method?

Reply:
I definitely could use it here, but I don't think it'll help much. This method gets all tags, ingredients, and steps using a while loop because the number of each is unknown for each recipe. getParameterMap() would organize these parameters into a HashMap, but I'd still have to loop over it in the while loop, because it won't tell me how much of each parameter there is.

Reply:
I think getParameterMap() returns a map with key, value[]. In that case, you can easily know the number of values for each of the parameter. Now, the issue I see is that if you have different parameter name for each (say) step, then it is hard to obtain that.
Have you considered having same parameter name, but with several values possible?

Reply:
I though about this approach, but I thought it might be easier to have multiple values for each parameter name in the next iteration of the feature. For example, in the next iteration, ingredients will have a string description, double amount, and string unit name (like cups, tablespoons). My plan was to have this data stored as multiple parameters on parameter name ingredient0, so the query string might have something like ingredient0=flour,2.5,cups&ingredient1=sugar,1,cups, etc.

Reply:
In this version, I'll probably use getParameterMap to make things easier.